---
title: "Getting started with Galaxy"
---

## A short introduction to Galaxy

To get started with the Galaxy platform, complete the following tutorial. It takes about 40 minutes and will familiarize you with the interface and the most important Galaxy terms. You will also upload your first data.

:::: hands-on
<hands-on-title>First steps with Galaxy</hands-on-title>

Go through and finish the tutiral "A short introduction to Galaxy"!\
The tutorial is available in several languages:

[![English](assets/gb.svg){.flag width="25px"} Galaxy Introduction Tutorial (GTN)](https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html)

[![Êó•Êú¨Ë™û](assets/jp.svg){.flag width="25px"} Galaxy Introduction Tutorial (GTN, automatic translation)](https://training-galaxyproject-org.translate.goog/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html?_x_tr_sl=en&_x_tr_tl=ja&_x_tr_hl=en&_x_tr_pto=wapp)

::: reference
<reference-title>Reference</reference-title> This tutorial is from the Galaxy Training Network (GTN): [*Introduction to Galaxy*](https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html) Lizenz: [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/).
:::
::::

## Starting with training data

Once the tutorial is complete, we can begin with our own data. I hope you enjoy it.\

![](assets/get_started_atgc.png){width="20%"} Let's get started!

## Create a new history

Now that the tutorial is complete, we can plan the work with our own data. To do this, we create a new, empty history. Then we rename the history to **Data NCBI**

![](png/history_new.PNG){width="80%"}

## Rename the new history

After creating the new history, we rename it. With the new name, it can be clearly identified and cannot be confused with previous analyses.

![](png/history_name.PNG){width="80%"}

## Download sequencing data

The sequence data that provided here is taken from one of my publications and is intended to serve as an example. The sequence data has been published on NCBI SRA and is freely available. The sequenced sample is a large dsDNA virus and the data is small in size. The small size makes it perfect, easy and quick to handle.

::: reference
<reference-title>Reference</reference-title>

| Isolate  |  BioSample   | NCBI SRA (Illumina) | NCBI SRA (Nanopore) |
|:--------:|:------------:|:-------------------:|:-------------------:|
| BmNPV-My | SAMN18849911 |     SRR14313723     |     SRR26992684     |

Wennmann, J.T., Lim, F.S., Senger, S., Gani, M., Jehle, J.A., Keilwagen, J (2024). Haplotype determination of the Bombyx mori nucleopolyhedrovirus by Nanopore sequencing and linkage of single nucleotide variants. Journal of General Virology. 105 (5), 001983. \[<https://doi.org/10.1099/jgv.0.001983>\]
:::

Please download the sequence data directly and save it temporarily on your computer. We will then upload it directly to your newly created Galaxy history ‚ÄúData NCBI.‚Äù

::: hands-on
<hands-on-title>Downloading sequencing data</hands-on-title>

1.  Follow this link to download the Illumina data and save it on your computer. You should download the forward and reverse reads in two separate files:\
    [üíæ \[Illumina forward reads\]](https://zenodo.org/records/17054180/files/SRR14313723_forward_illumina.fastqsanger.gz)\
    [üíæ \[Illumina reverse reads\]](https://zenodo.org/records/17054180/files/SRR14313723_reverse_illumina.fastqsanger.gz)

2.  Follow the link to download the Nanopore reads:\
    [üíæ \[Nanopore reads\]](https://zenodo.org/records/17054180/files/SRR26992684_nanopore.fastqsanger.gz)
:::

You should now have three files saved on your computer.

## Upload your data to Galaxy

Sequence data is usually obtained directly in fastq or fastqsanger format. It is then uploaded to the Galaxy platform. We will do this in the next step.

::: hands-on
<hands-on-title>Upload your data to Galaxy</hands-on-title>

1.  You have the three files.\
2.  Go to the folder where you saved the files.\
3.  Go to usegalaxy.eu and click on ‚ÄúUpload.‚Äù\
4.  Drag your files into the upload window (drag and drop).\
5.  Click on ‚ÄúStart.‚Äù The upload will begin.\
6.  Once all files have been uploaded, click on ‚ÄúClose.‚Äù

![](png/upload_data.gif){width="100%"}

Now you have to wait until all files are completely uploaded. This may take a moment. The upload is complete when the files appear green in the history.

![](png/file_status.PNG){width="80%"}
:::

## Status of files in your history

Files can have four statuses in your history. Only green files and files being processed can be worked on further.

| Color     | Status             | Meaning                                     |
|------------------|--------------------|----------------------------------|
| üü© Green  | Ready              | File is ready to use                        |
| üüß Orange | Running/Processing | File is being processed (job still running) |
| ‚¨ú Gray   | Queued             | File is waiting to run                      |
| üü• Red    | Error              | File has failed (an error occurred)         |

## Run your first Galaxy tools

Once the data has been uploaded, we want to do something with it. We want to analyse it, not just look at it. But before we analyse it, it's important that we get to know the data well. It's important to know whether the sequencing worked well. We also want to know how good the quality of the sequencing is.

There are two tools we can use for Illumina and Nanopore data.

::: hands-on
<hands-on-title>Practice tool selection</hands-on-title>

Please practice to select some tools. Go to web interface of usegalaxy.eu and click on tools. In the tools menu you can search for tools you would like to apply to your data. Please search for two tools and select them:

-   falco
-   nanoplot

![](png/tool_selection.gif){width="100%"}
:::

## Quality check of Nanopore reads

At first we start the tool `nanoplot`{.tool} on the Nanopore data. The file's name should be `SRR26992684_nanopore.fastqsanger.gz` and it is the input for the tool `nanoplot`{.tool}.

::: hands-on
<hands-on-title>Start tool nanoplot on Nanopore data</hands-on-title>

![](png/tool_nanoplot.png){width="100%"}
:::

## Quality check of Illumina reads

Next, we use the `falco`{.tool} tool, which we run on the forward file of the Illumina data. Here we chose the file `SRR14313726_forward_illumina.fastqsanger.gz`.

::: hands-on
<hands-on-title>Start tool falco on Illumina data</hands-on-title>

![](png/tool_falco.png){width="100%"}
:::

## The correct files for the right tools

Not all files can be read, processed or analysed by all tools. You learn this over time and with increasing experience.

::: warning
<warning-title>Chose the correct file!</warning-title>

A common mistake is selecting the wrong files for the wrong tool. Falco is for short reads (Illumina data) and Nanoplot is for long reads. Therefore, the following combination must be selected:

-   `falco`{.tool} = `SRR14313726_forward_illumina.fastqsanger.gz`\
-   `falco`{.tool} = `SRR14313726_reverse_illumina.fastqsanger.gz`
-   `nanoplot`{.tool} = `SRR26992684_nanopore.fastqsanger.gz`
:::

## Reviewing the output and evaluating the quality

To view the results of the tools, click on the eye icon next to the outputs. Make sure you select the correct output files. They are called `html report` for the `nanoplot`{.tool} tool and `webpage` for the `falco`{.tool} tool.

::: hands-on
<hands-on-title>Let's discuss the results together!</hands-on-title>

Click on the eye of one of the files. The results should open in the center panel.

![](png/tool_eye_falco_nanoplot.png){width="100%"}
:::

![](assets/get_end_atgc.png){width="20%"} The results are discussed together in the group.

::: comment
<comment-title>Self-study to interpret the results</comment-title>

If you would like to learn how to evaluate and analyse the output of `faclo`{.tool} (or `fastqc`{.tool}, which is the predecessor of `faclo`{.tool}) and `nanoplot`{.tool} yourself, you can do so in the following tutorial:  
[Galaxy Training Tutorial - Quality Control](https://training.galaxyproject.org/topics/sequence-analysis/tutorials/quality-control/tutorial.html).

Enjoy reading!
:::
